substrate/client/network/sync/src/strategy/chain_sync.rs

```diff
@@ -468,130 +476,160 @@ where
 
 	/// Notify syncing state machine that a new sync peer has connected.
 	pub fn add_peer(&mut self, peer_id: PeerId, best_hash: B::Hash, best_number: NumberFor<B>) {
-		match self.add_peer_inner(peer_id, best_hash, best_number) {
-			Ok(Some(request)) =>
-				self.actions.push(ChainSyncAction::SendBlockRequest { peer_id, request }),
-			Ok(None) => {},
-			Err(bad_peer) => self.actions.push(ChainSyncAction::DropPeer(bad_peer)),
-		}
+		self.peers.insert(
+			peer_id,
+			PeerSync {
+				peer_id,
+				common_number: Zero::zero(),
+				best_hash,
+				best_number,
+				state: PeerSyncState::New,
+			},
+		);
 	}
 
-	#[must_use]
-	fn add_peer_inner(
+	/// Process new peers assigning proper states and initiating requests.
+	fn handle_new_peers(
```

Comment:
It's unclear to me why this function is needed and why can't the new peers be handled in `ChainSync::block_requests()`

Reply:
I'll look if this can be simplified.